From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Mon, 18 Jun 2018 23:12:58 +0200
Subject: [PATCH] {For LLVMLinux} drivers/gpu/drm/amd/amdgpu: do not use ->
 notation in container_of

clang complains (struct amdgpu_uvd has a field "struct amdgpu_uvd_inst
inst[2];")

    drivers/gpu/drm/amd/amdgpu/amdgpu_uvd.c:1148:52: error: expected ')'
                    container_of(work, struct amdgpu_device, uvd.inst->idle_work.work);
                                                                     ^
    drivers/gpu/drm/amd/amdgpu/amdgpu_uvd.c:1148:3: note: to match this '('
                    container_of(work, struct amdgpu_device, uvd.inst->idle_work.work);
                    ^
    include/linux/kernel.h:988:21: note: expanded from macro 'container_of'
            ((type *)(__mptr - offsetof(type, member))); })
                               ^
    include/linux/stddef.h:17:32: note: expanded from macro 'offsetof'
    #define offsetof(TYPE, MEMBER)  __compiler_offsetof(TYPE, MEMBER)
                                    ^
    include/linux/compiler-gcc.h:170:20: note: expanded from macro '__compiler_offsetof'
            __builtin_offsetof(a, b)
                              ^
    drivers/gpu/drm/amd/amdgpu/amdgpu_uvd.c:1147:24: error: initializing
    'struct amdgpu_device *' with an expression of incompatible type
    'void'
            struct amdgpu_device *adev =
                              ^
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_uvd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_uvd.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_uvd.c
index 691c3b34f9ac..a3ffbb04a0aa 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_uvd.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_uvd.c
@@ -1145,7 +1145,7 @@ int amdgpu_uvd_get_destroy_msg(struct amdgpu_ring *ring, uint32_t handle,
 static void amdgpu_uvd_idle_work_handler(struct work_struct *work)
 {
 	struct amdgpu_device *adev =
-		container_of(work, struct amdgpu_device, uvd.inst->idle_work.work);
+		container_of(work, struct amdgpu_device, uvd.inst[0].idle_work.work);
 	unsigned fences = 0, i, j;
 
 	for (i = 0; i < adev->uvd.num_uvd_inst; ++i) {
-- 
